source/configuration_manual/mail_location/index.rst

```diff
@@ -129,34 +129,32 @@ Key                Value Description
                    directories are stored. The default is empty unless
                    otherwise described in the mailbox format pages.
 
+``DIRNAME``        Specifies the directory name used for mailbox directories,
+                   or in the case of mbox specifies the mailbox message file
+                   name.
+
+                   .. note:: ``DIRNAME`` is not used for index or control
+                             directories, consider using ``FULLDIRNAME``
+                             instead.
+
+                   .. deprecated:: v2.4.0
+
 ``FULLDIRNAME``    Specifies the directory name used for mailbox, index, and
                    control directory paths. See the individual mailbox format
-                   pages for further information. This key replaced the
-                   deprecated ``DIRNAME`` key.
+                   pages for further information.
+
+                   .. note:: Bug: Before v2.4.0/v3.0.0 when using
+                             ``FULLDIRNAME`` together with ``INDEX``,
+                             ``INDEXPVT``, ``INDEXCACHE`` or ``CONTROL``
+                             options renaming the parent folder causes its
+                             child folders to lose all messages.
```

Comment:
This could be changed to similar sentences as in the release note:

Before v2.3.19 folder renaming was broken when using FULLDIRNAME together with INDEX, INDEXPVT, INDEXCACHE or CONTROL. Renaming a parent folder lost the index/control files for all of its child folders, which with some mailbox formats lost the mails until force-resync was used.